### PR TITLE
Fix/users controller

### PR DESCRIPTION
**Pull request by dbyington**
This is tests and code for the users controller.
As an aside two middlewares were created as dependencies.

### CODE REVIEW

**Review by angelicavalenta — APPROVED** [review]

File: middlewares/authentication.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const fetch = require('node-fetch');
+const request = require('request-promise');
+const User = require('../models/user.model');
+
+const User = require('../models/user.model');
+const db = require('../db');
+
+const getAccessTokenUrl = 'https://api.instagram.com/oauth/access_token';
+const CLIENT_ID = '38553e7b653747dfa48428b052c7369d';
+const CLIENT_SECRET = 'b5466dfe3a7f470180f8148345bb67db';
```
Comment (arol) [inline middlewares/authentication.js]: 🙌

File: middlewares/authentication.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+const _getAccessToken = (ctx) => {
+  const form = {
+    client_id = CLIENT_ID,
+    client_secret = CLIENT_SECRET,
+    grant_type = 'authorization_code',
+    redirect_uri = 'https://me/authorize',
```
Comment (arol) [inline middlewares/authentication.js]: Add this on the environment variables (`dotenv`?)

File: middlewares/user-serializer.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+'use strict';
```
Comment (arol) [inline middlewares/user-serializer.js]: This is an utility (utils), not a middleware.

Comment (arol) [reply middlewares/user-serializer.js]: Or you can place them (this and future serializers) on the a new serializers folder.

File: models/media.model.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+const Schema = mongoose.Schema;
+
+const mediaSchema = new Schema({
+  id: Number,
+  url: String,
+  link: String,
+  likes: Number,
+  comments: Number,
+  tags: Array
+});
```
Comment (arol) [inline models/media.model.js:11]: Create the model and export it.

File: test/controllers.test.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+describe('Users', function() {
+  describe('GET /authorize', function() {
+    it('should return the user object', async function(){
+      let ctx = Object.assign({}, mocks.ctx, {state: {accessToken: 'ACCESS_TOKEN'}})
+      await users.handleAuthorizeUser(ctx);
+      ctx.body.should.eql(mocks.authUser);
+    });
+    it('should return the access token', async function(){
+      let ctx = Object.assign({}, mocks.ctx, {state: {accessToken: 'ACCESS_TOKEN'}})
```
Comment (arol) [inline test/controllers.test.js]: Same preparation as the last case: use `beforeAll`

File: test/controllers.test.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+  describe('GET /me', function() {
+    it('should return the user object', async function(){
+      let ctx = Object.assign({}, mocks.ctx, {state: {accessToken: 'ACCESS_TOKEN'}})
+      await users.handleUserGet(ctx);
+      ctx.body.should.eql(mocks.getUser);
+    });
+    it('should not return an access token', async function(){
+      let ctx = Object.assign({}, mocks.ctx, {state: {accessToken: 'ACCESS_TOKEN'}})
+      await users.handleUserGet(ctx);
```
Comment (arol) [inline test/controllers.test.js]: Same here (`beforeAll`)

**Review by arol — CHANGES_REQUESTED** [review]

**Comment by dbyington** [comment]
Changes made in commit eebb7f0

**Review by arol — APPROVED** [review]

**Pull request merged**